plugins/catalog-backend/src/processing/DefaultCatalogProcessingEngine.ts

```diff
@@ -295,7 +295,7 @@ export class DefaultCatalogProcessingEngine {
                 });
               oldRelationSources = new Map(
                 previous.relations.map(r => [
-                  `${r.source_entity_ref}:${r.type}`,
+                  `${r.source_entity_ref}:${r.type}:${r.target_entity_ref}`,
```

Comment:
nit: do we want to have something like
```suggestion
                  `${r.type}:${r.source_entity_ref}->${r.target_entity_ref}`,
```
here? that's a bit more clear wrt directionality and type

Reply:
I like it!